bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/auth/client/oauth2/OAuthFactory.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.auth.client.oauth2;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * The OAuth Factory interface
+ *
+ * @author Michael Bock - Initial Contribution
+ * @author Gary Tse - ESH Adaptation
+ */
+@NonNullByDefault
+public interface OAuthFactory {
+
+    /**
+     * Creates a new oauth service. Use this method only once to obtain a handle and store
+     * this handle for further in a persistent storage container.
+     *
+     * @param tokenUrl the token url of the oauth provider. This is used for getting access token.
+     * @param authorizationUrl the authorization url of the oauth provider. This is used purely for generating
+     *            authorization code/ url.
+     * @param clientId the client id
+     * @param clientSecret the client secret (optional)
+     * @param scope the desired scope
+     * @param supportsBasicAuth whether the OAuth provider supports basic authorization or the client id and client
+     *            secret should be passed as form params. true - use http basic authentication, false - do not use http
+     *            basic authentication, null - unknown (default to do not use)
+     * @return a handle to the oauth service
+     */
+    String createOAuthClientService(String tokenUrl, @Nullable String authorizationUrl, String clientId,
+            @Nullable String clientSecret, @Nullable String scope, @Nullable Boolean supportsBasicAuth);
+
+    /**
+     * Gets the oauth service for a given handle
+     *
+     * @param handle the handle to the oauth service
+     * @return the oauth service, or null of the handle does not exist
+     * @throws OAuthException if the handles does not exist
```

Comment:
Typo: "handles" -> "handle"

Reply:
This actually contradicts the null return value which states "null of the handle does not exist"